sgt-back-end/index.js

```diff
@@ -0,0 +1,209 @@
+const express = require('express');
+const app = express();
+
+app.use(express.json()); // middleware
+
+const pg = require('pg');
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+// GET ONE ID
+app.get('/api/grades/:gradeId', (req, res, next) => {
+
+  const gradeId = Number(req.params.gradeId);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    res.status(400).json({
+      error: '"gradeId" must be a positive integer.'
+    });
+    return;
+  }
+
+  const sql = `
+    select "gradeId",
+           "name",
+           "course",
+           "score",
+           "createdAt"
+      from "grades"
+     where "gradeId" = $1
+  `;
+
+  const params = [gradeId];
+
+  db.query(sql, params)
+    .then(result => {
+      const grade = result.rows[0];
+      if (!grade) {
+        res.status(404).json({
+          error: `Cannot find grade with "gradeId" ${gradeId}.`
+        });
+      } else {
+        res.json(grade);
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: `An unexpected error occurred.`
+      });
+    });
+});
+
+// GET ALL
+app.get('/api/grades', (req, res, next) => {
+
+  const sql = `
+    select *
+      from "grades"
+  `;
```

Comment:
In web applications, it's better to specify which columns you need than to do a blanket column select (`'*'`).